### DISCUSSION TITLE
1Password (1PUX) Import Kind-of-Working

### DISCUSSION THREAD

**Discussion by quella** [dani-garcia/vaultwarden#3110] [Q&A]
First, thank you for all your hard works and getting the product where it is today.  I'm looking at moving from 1Password subscription model to a self-hosted model.  I am running the latest docker Vaultwarden server version (2022.12.0) on Docker and R Raspberry Pi.

I am Attempting to use the web UI import feature (Tools -> Import Data) and providing the data from a 1Password v8.9.11 1PUX export file.  Much of the import seems to work, but quite a few accounts did not get imported and unsure why.  Many of the ones I have seen so far had files attached, but others that has files imported ok (lacking the file which is OK).  

Was not sure if I'm missing something here or if 1Password may have modified the format of their 1PUX.  Today, when doing an export from 1Password, it only provides two options.  1PUX if you want to export EVERYTHING and CSV is you want to export only Login/Password (no OTP, Notes, files, etc.)

Wondering how others may have solved this issue and I like the self-hosted tool and see it as a viable solution in the long run.

**Comment by BlackDex** [comment]
My guess is that those are or have notes which are larger then supported. You should be able to see error message in the logs regarding this. Also, not sure if Bitwarden does import other items then logins for some specific formats.

Since import/export are all client based items, i suggest to check the bitwarden/clients repo if there are any reports there.

**Reply by quella** [reply]
@BlackDex Thanks for the follow-up on my question around importing 1Password 1PUX files and some missing logins.  The import i'm using is from the Docker Web UI Tools Menu, so I did not assume this was a Bitwarden client issue.  During the import, there is no feedback or errors that appear.  Where in the Docker deployment of Vaultwarden would the logs be visible?  Under reports, or in the docker container itself?

**Reply by BlackDex** [reply]
The web-ui/web-vault is a client also 😉 . Vaultwarden only makes some minor changes so that it works correctly with the Vaultwarden backend, other than that, it is a 1:1 copy.

All the importing/exporting is done client side. That is what converts the formats to a Bitwarden json format, which has everything encrypted before sending it to the backend.

The logs would be visible either in the file configured via `LOG_FILE`, or via docker logs. Not sure what you are using, but via the CLI it would be as simple as `docker logs vaultwarden`.

**Reply by quella** [reply]
@BlackDex Thanks again for the support and my better understanding.  I will try a new import and have a look at the logs.  I'm running Docker on a RPI and your command worked when I issued it from the CLI.  I'll do some playing and try to find what might be causing the import to be missing some logins, etc.
